core/templates/components/common-layout-directives/navigation-bars/top-navigation-bar.component.ts

```diff
@@ -344,6 +357,13 @@ export class TopNavigationBarComponent implements OnInit, OnDestroy {
     }, 0);
   }
 
+  ngAfterViewInit(): void {
+    // Wait for userIsLoggedIn = true.
+    setTimeout(() => {
+      this.showNumberOfUnreadfeedback();
+    }, 1010);
+  }
```

Comment:
can you please explain these changes i.e. why setTimeout & is there any alternative that we can use in place of setTimeout?

Reply:
I am fetching the number of unread feedback messages only when the user is logged in.
i.e 
If (UserIsLoggedIn){
//fetch
}
To execute the function I need the UserIsLoggedIn to true.
But it takes some time.
To resolve this issue first I have used **ngAfterViewInit**
But it does not work. So I need to use setTimeOut to fix this.
I think this is a way that I can do it.
This way is also used at other place in the code base.

Reply:
can you please try to handle this situation through Promise or EventEmitters/Observables? `this is not an explanation of doing things This way is also used at other places in the code base so you used it here. try to justify it technically or logically`

Reply:
can you please point out the file where this is used?

Reply:
Please see here for the same use 
1 (https://github.com/oppia/oppia/blob/develop/core/templates/pages/exploration-editor-page/statistics-tab/charts/pie-chart.component.ts#L88)
2(https://github.com/oppia/oppia/blob/develop/extensions/objects/templates/graph-editor.component.ts#L44)
3 https://github.com/oppia/oppia/blob/develop/extensions/interactions/CodeRepl/directives/oppia-interactive-code-repl.component.ts#L170

Reply:
The use of setTimeout needs to be minimised because it can result in unexpected change detection related issues. Have you tried Shivam's recommendation above regarding using EventsEmitters / Observables? We probably have examples on how to use those in the codebase.

Reply:
Removed this and use the different approach here (https://github.com/oppia/oppia/pull/18350/files#diff-05f6faf0ee85d42e18432938fbdfb4a2ee8783b4ccdea0e8a87bcb3775071ce5R276)
Thanks!